addons/sale_crm/models/crm_team.py

```diff
@@ -42,3 +42,52 @@ def _extra_sql_conditions(self):
         if self.use_opportunities and self._context.get('in_sales_app'):
             return "AND state in ('sale', 'done', 'pos_done')"
         return super(CrmTeam,self)._extra_sql_conditions()
+
+    def _merge_leads(self, leads, values):
+        """ Remove opportunities or leads from ["leads_dups_dict"] that
+        have sale orders and the company of the sale order is not the same
+        as the company of the head lead, its sale order, or current company.
+
+        :param leads: recordset of leads to assign to current team;
+        :param values: dictionary in the following form
+                       {
+                            'lead_duplicates': crm.lead(),
+                            'leads_assigned': crm.lead(),
+                            'leads_dups_dict': {crm.lead(): crm.lead()}
+                       };
+        :return: Opportunities or leads that have sale orders, and the company
+        of the sale order is not the same as the company of the head lead, its
+        sale order, or current company.
+        (
+            {'leads_with_different_company': opportunities}
+        )
+        """
+        opportunities = values['lead_duplicates']._sort_by_confidence_level(reverse=True)
+        if opportunities.order_ids \
+           and (len(opportunities.order_ids.company_id) > 1 or len(opportunities.company_id) > 1):
```

Comment:
Yes, it matters because in a case like this, there is only one sales order, which means that we have only one company in sales orders, while in opportunities there are more than one company, and when CRON tries to merge opportunities, it crashes because the sale order company is different than the head opportunity.

Reply:
Ah yes, of course. We don't necessarily merge into the team of the sales order. Got it :)